doc: release_notes: Updates from applications (#4904)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-latest.rst

  * Reworked the :ref:`NCP sample <zigbee_ncp_sample>` to work with the simple gateway application.
  * Moved the `NCP Host documentation`_ from the `ZBOSS NCP Host`_ package to the same location as the `external ZBOSS development guide and API documentation`_.

nRF Desktop
-----------

* Fixed boot reports on USB backend.

> * Use common modules from CAF library.

Reviewer: ```suggestion
* Adapted the application to use common modules from the :ref:`lib_caf` library.
```
@emob-nordic , can you add release notes for the CAF-related tasks?

> * Minor bugfixes.

Reviewer: ```suggestion
* Fixed minor bugs.
```

Common
======

The following changes are relevant for all device families.

Edge Impulse
------------

* Added the example application integrating edge impulse wrapper library with sensor sampling.
  The code is located in :file:`ncs/nrf/applications/machine_learning`.
* Sample reference to a public pre-trained model.

Pelion
------

* Added integration of Pelion device management library into the |NCS|.
  For more information refer to :ref:`ug_pelion`.

> 

Reviewer 2: Added an example application…

Author: changed

  * Reworked the :ref:`NCP sample <zigbee_ncp_sample>` to work with the simple gateway application.
  * Moved the `NCP Host documentation`_ from the `ZBOSS NCP Host`_ package to the same location as the `external ZBOSS development guide and API documentation`_.

nRF Desktop
-----------

* Fixed boot reports on USB backend.
* Use common modules from CAF library.
* Minor bugfixes.

Common
======

The following changes are relevant for all device families.

Edge Impulse
------------

> * Added the example application integrating edge impulse wrapper library with sensor sampling.
>   The code is located in :file:`ncs/nrf/applications/machine_learning`.

Reviewer: ```suggestion
* Added the :ref:`nrf_machine_learning_app` application that integrates the Edge Impulse wrapper library with sensor sampling.
```
I don't recall providing mention where the code is in previous release notes.

> * Sample reference to a public pre-trained model.

Reviewer: Not sure what this is about. @MarekPieta ?

Author: That you don't have to train model yourself as there is a default model added to config.

Pelion
------

> * Added integration of Pelion device management library into the |NCS|.
>   For more information refer to :ref:`ug_pelion`.

Reviewer: ```suggestion
* Added the integration of Pelion Device Management library, available as one of the external submodule repositories in the |NCS|.
  For more information, see :ref:`ug_pelion`.
```

> 

Reviewer: Remove blank line.

> * Added an example application utilizing the library.
>   The code is located in :file:`ncs/nrf/applications/pelion_client`.

Reviewer: ```suggestion
* Added the :ref:`pelion_client` application that showcases the usage of Pelion Device Management library.
```

Common Application Framework (CAF)
----------------------------------

> * Some of the application modules of nRF Desktop application were migrated and reused by other applications.
>   For more information refer to :ref:`lib_caf`.

Reviewer: ```suggestion
* Migrated some of the application modules of nRF Desktop application to :ref:`lib_caf` for reuse by other applications.
```
